sktime/forecasting/multiplexer.py

```diff
@@ -0,0 +1,173 @@
+    def __init__(
+        self,
+        component_estimators: list,
+        selected_estimator=None,
+    ):
+
+        """
+        MultiplexerForecaster facilitates a framework for performing
+        model selection process over different model classes.
+        It should be used in conjunction with ForecastingGridSearchCV
+        to get full utilization.
+
+        Single use of MultiplexerForecaster with component_estimators
+        and selected_estimator parameter specified,
+        works just like the selected component.
+        It does not provide any further use in that case.
+
+        When used with ForecastingGridSearchCV, MultiplexerForecaster
+        provides an ability to compare different model class
+        performances with each other, just like a model tournament.
+        When ForecastingGridSearchCV is fitted with a MultiplexerForecaster,
+        returned value for the selected_estimator argument of best_params_
+        attribute of ForecastingGridSearchCV, gives the best
+        performing model class among given models provided in component_estimators.
+
+        Parameters
+        ----------
+        component_estimators : list
+            List of (forecaster names, forecaster objects)
+            MultiplexerForecaster switches between these forecasters
+            objects when used with ForecastingGridSearchCV to
+            find the optimal model
+        selected_estimator: str
+            An argument to make a selection among component_estimators.
+            MultiplexerForecaster uses selected_estimator
+            to choose which component to fit.
+            Important for using with ForecastingGridSearchCV as a
+            hyperparameter.
+
+        Attributes
+        ----------
+        _forecaster : Sktime forecaster
+            forecaster that MultiplexerForecaster will currently
+            forecast with.
+        _forecaster_fit_params: dict
+            Fit params for the forecaster that is
+            going to be used by the forecaster in the
+            _forecaster
+        """
+
+        self.component_estimators = component_estimators
+        self.selected_estimator = selected_estimator
+        self._check_components()
+        self._forecaster_fit_params = None
+
+        super(MultiplexerForecaster, self).__init__()
+
+    def _check_components(self):
+        if not isinstance(self.component_estimators, list):
+            raise Exception(
+                "Please provide a list " "for component_estimators composed of tuples"
+            )
+
+        for component in self.component_estimators:
+            name, estimator = component
+            if not isinstance(estimator, _SktimeForecaster):
+                raise Exception(
+                    "Each component has to be an "
+                    "sktime forecaster object. "
+                    "Please check {} with name:{}".format(estimator, name)
+                )
+
+    def _check_fit_params(self, fit_params):
```

Comment:
is this from scikit-learn? how do they do this?

Reply:
I removed the _update_fit_params and moved all the content in to _check_fit_params following the sklearn's practice
given here: 

https://github.com/scikit-learn/scikit-learn/blob/95119c13a/sklearn/pipeline.py#L37

---

Kutay Koralturk kutay.koralturk@daimler.com, Mercedes-Benz AG on behalf of Daimler TSS GmbH.
https://github.com/Daimler/daimler-foss/blob/master/LEGAL_IMPRINT.md